### DISCUSSION TITLE
Duda sobre TP1 - Contador Circular (Ej 11)

### DISCUSSION THREAD

**Discussion by alengeor** [INGCOM-UNRN-P1/cursada-2023#35] [Preguntas y respuestas] [answered]
Buenas tardes a todos! Espero que anden bien.

No termino de comprender la consigna del punto 11 del TP1, contador circular:

![image](https://github.com/INGCOM-UNRN-P1/cursada-2023/assets/110181852/2661804a-6592-4a13-9afe-2e95c9063e0d)

Traté de implementar una función que vaya retornando, uno a uno, _los números comprendidos entre i y límite_, pero no me funcionó. 

Por ejemplo, **i = 30, limite = 35**. Yo esperaba que mi función retorne **31, 32, 33, 34** y **0**.

En esencia, quise hacer que una funcion retorne mas de una vez, lo que según entiendo, no es posible.

Es por eso que comienzo a sospechar que no comprendí la consigna jajajaja.

Cualquier tip es bienvenido :D

Buen finde!!!

**Comment by JeanFranco99** [comment, marked as answer]
Claro, es eso lo que tendria que hacer.
Por ejemplo, si tu i= 30 y limite=35, tendria que mostrarse. Un tip es que agregues por defecto o que se ingrese por teclado, la cantidad de veces que va a mostrarse (ejemplo 15 veces)
30
31
32
33
34
0
30
31
32
33
34
0
30
31

**Reply by martinvilu** [reply]
Usa el mismo ejemplo que te envie Jean;

Es más simple.
La idea de esta función es algo así.

Límite en 3, número en 0. y cada renglón es una llamada a la función.

0
1
2
0
1
2
0
1
2

Para darle un uso razonable a esta función, es necesario dos lazos. uno que sea la "cantidad de vueltas" y otro para hacer las vueltas con la función de contador circular

**Reply by alengeor** [reply]
Hmmm, interesante.
Logré llegar a esto, que conceptualmente se acerca a lo que explican.

Pero no son retornos, son prints dentro de la función:


<details>
  <summary>Spoiler</summary>
 
```C
#include <stdio.h>

int contador_circular(int i, int limite)
{
    while(i < limite)
    {
        printf("\n%i", i);
        i++;
    }
    printf("\n0");
}

int main()
{
    int numero_in;
    int limite_in;
    int cantidad_vueltas;
    
    printf("Ingrese un numero entero:");
    scanf("%i", &numero_in);
    
    printf("Ingrese un limite, mayor al numero ingresado:");
    scanf("%i", &limite_in);
    
    printf("Ingrese la cantidad de vueltas:");
    scanf("%i", &cantidad_vueltas);
    
    int j = 0;
    
    while(j <= cantidad_vueltas)
    {
        contador_circular(numero_in, limite_in);
        j++;
    }
}
```
  
</details>

**Reply by alengeor** [reply]
<details>
  <summary>Spoiler</summary>
  
Esta es otra alternativa:
  
```C
#include <stdio.h>

int contador_circular(int i, int limite)
{
    while(i < limite)
    {
        printf("\n%i", i);
        i++;
    }
      return(0);
}

int main()
{
    int numero_in;
    int limite_in;
    int cantidad_vueltas;
    
    printf("Ingrese un numero entero:");
    scanf("%i", &numero_in);
    
    printf("Ingrese un limite, mayor al numero ingresado:");
    scanf("%i", &limite_in);
    
    printf("Ingrese la cantidad de vueltas:");
    scanf("%i", &cantidad_vueltas);
    
    int j = 0;
    
    while(j <= cantidad_vueltas)
    {
        contador_circular(numero_in, limite_in);
        j++;
    }
    
```
</details>

**Reply by alengeor** [reply]
Se me hace que la segunda opción se arrima mas a lo que pide la consigna. Además que como salida tiene lo mismo que lo que compartió el profe Martín:

![image](https://github.com/INGCOM-UNRN-P1/cursada-2023/assets/110181852/62e59bbb-9c9e-4d5f-861b-fafd9d1cdad2)

**Reply by martinvilu** [reply]
Los prints, hacen que la función no cumpla con su objetivo, que es la de mantener `numero` en entre 0 y `limite`.
